ouroboros-consensus/docs/diffusion-pipelining-design/block-diffusion-pipelining.tex

```diff
@@ -0,0 +1,554 @@
+\end{enumerate}
+
+On a stylized time-line (with time running from left to right), we can depict
+this as:
+
+\begin{center}
+\begin{tikzpicture}
+\begin{scope}[thick, xscale=0.9]
+
+\draw (0,0) rectangle (2,1);
+\draw[fill=lightgray] (2,0) rectangle (3,1);
+\draw (3,0) rectangle (7,1);
+\draw[fill=lightgray] (7,0) rectangle (10,1);
+\draw (10,0) rectangle (12,1);
+\draw (13,0) rectangle (17,1);
+
+\draw (1,  0.5) node {1.};
+\draw (2.5,0.5) node {2.};
+\draw (5,  0.5) node {3.};
+\draw (8.5,0.5) node {4.};
+\draw (11, 0.5) node {5.};
+\draw (15, 0.5) node {6.};
+
+\draw[->] (0, 1.5) -- (17, 1.5);
+\draw     (8.5, 1.5) node[anchor=south] {Time};
+
+\end{scope}
+\end{tikzpicture}
+\end{center}
+
+Of course, this represents a single hop in the broader picture of block propagation and in a network there will typically
+be 5 -- 7 such hops between a given block producer, and the last block producer in the path. In the current design we 
+fully validate blocks before relaying them, so the overall time-line looks like this:
+
+\begin{center}
+\begin{tikzpicture}
+\begin{scope}[thick, xscale=0.355, yscale=0.5]
+
+\begin{scope}
+  \draw[fill=lightgray] (7,0) rectangle (10,1);
+  \draw (10,0) rectangle (12,1);
+  \draw (13,0) rectangle (17,1);
+  \draw (17, 0.5) node[anchor=west] {Block forger node};
+\end{scope}
+
+\begin{scope}[xshift=10cm, yshift=-2cm]
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (10,1);
+  \draw (10,0) rectangle (12,1);
+  \draw (13,0) rectangle (17,1);
+\end{scope}
+
+\begin{scope}[xshift=20cm, yshift=-4cm]
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (10,1);
+  \draw (10,0) rectangle (12,1);
+  \draw (13,0) rectangle (17,1);
+  \draw (4, -0.5) node[anchor=north east] {Intermediate relay nodes};
+\end{scope}
+
+\begin{scope}[xshift=30cm, yshift=-6cm]
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (10,1);
+  \draw (10,0) rectangle (12,1);
+  \draw (13,0) rectangle (17,1);
+\end{scope}
+
+\begin{scope}[xshift=40cm, yshift=-8cm]
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (10,1);
+  \draw (0, 0.5) node[anchor=east] {Last block producer in path};
+\end{scope}
+
+\draw[->] (7, 2) -- (50, 2);
+\draw     (23.5, 2) node[anchor=south] {Time};
+
+\end{scope}
+\end{tikzpicture}
+\end{center}
+
+\pagebreak
+
+\section{The proposed block relaying scheme}
+By propagating forward a block as soon as it has arrived locally (and its header is validated), the block 
+validation time is overlapped with the propagation time and the overall timeline becomes materially shorter.
+
+\begin{center}
```

Comment:
I think these pictures are hard to understand. In particular, the pictures in this chapter do not show that a node is simultaneously validating the block while also transmitting the header and body. I _think_ we're only seeing the validation time and the part of the block transmission that ends after block validation completes (even though transmission might end before validation, yeah?). Relatedly: I think header transmission is not even shown (there are only 5 rectangles instead of 6 from the previous chatper's pictures).

Ideally, the picture could be enriched (eg with a thatched section implying an overlaid grey and white rectangles) (eg dividing each node's horizontal bar into three thinner lanes, one for header transmission, one for block validation, and one for block transmission -- the only rule is that block transmission comes after block validation; header transmission and block validation could and would likely begin simultaneously).

Reply:
yep I will try and add to this later